creator-node/test/ffmpeg.js

```diff
@@ -0,0 +1,94 @@
+const { segmentFile } = require('../src/ffmpeg')
+
+const path = require('path')
+const fs = require('fs')
+const assert = require('assert')
+
+describe('test segmentFile', () => {
+  // Create the segments directory to store segments in
+  before(() => {
+    const segmentsDirPath = path.join(__dirname, 'segments')
+    if (!fs.existsSync(segmentsDirPath)) {
+      try {
+        fs.mkdirSync(segmentsDirPath)
+      } catch (e) {
+        assert.fail(`Could not create dir at ${segmentsDirPath}`)
+      }
+    }
+  })
+
+  /**
+   * Given: improper params are passed in
+   * When: track segmenting occurs
+   * Then: an error is thrown
+   */
+  it('should throw an error if ffmpeg reads bad params', async () => {
+    try {
+      await segmentFile(null, null, {})
+      assert.fail('Should have thrown error with bad params')
+    } catch (e) {
+      console.error(e)
+      assert.ok(true)
+    }
+  })
+
+  /**
+   * Given: a corrupted track is present (image)
+   * Then: an error is thrown
+   * When: it is segmented
+  */
+  it('should throw an error if ffmpeg reads a bad track file (image)', async () => {
+    const fileDir = __dirname
+    const fileName = 'testTrackWrongFormat.jpg'
+
+    try {
+      await segmentFile(fileDir, fileName, {})
+      assert.fail('Should have thrown error when segmenting a bad track (image)')
+    } catch (e) {
+      console.error(e)
+      assert.ok(true)
```

Comment:
dont have to do it here, but it would be good to actually assert on the error message